functions/routes/ecom/modules/create-transaction.js

```diff
@@ -26,30 +42,175 @@ exports.post = ({ appSdk }, req, res) => {
+    if (installmentsNumber > 1) {
+      if (config.installments) {
+        // list all installment options
+        const { gateway } = addInstallments(amount, config.installments)
+        const installmentOption = gateway.installment_options &&
+          gateway.installment_options.find(({ number }) => number === installmentsNumber)
+        if (installmentOption) {
+          transaction.installments = installmentOption
+          finalAmount = transaction.installments.total =
+            Math.round(installmentOption.number * installmentOption.value * 100) / 100
+        } else {
+          installmentsNumber = 1
+        }
+      }
+    }
+
+    const payerIp = params.browser_ip
+    data = JSON.parse(Buffer.from(params.credit_card.hash, 'base64'))
+    data.metadata.risk.payer_ip = payerIp
 
-  // https://gist.github.com/luisbebop/ca87e04da04bcf662f732b1b6848d6ca#integration-
-  // https://infinitepay.io/docs#listar-wallets
-  const infinitepayAxiosConfig = {
-    headers: {
-      Authorization: config.infinitepay_api_key
+    data.customer = IPCustumer
+    data.payment = {
+      amount: Math.floor(finalAmount * 100),
+      installments: installmentsNumber,
+      payment_method: 'credit',
+      capture_method: 'ecommerce'
     }
+    const ipItems = []
+    items.forEach(item => {
+      ipItems.push({
+        id: item.sku || item.variation_id || item.product_id,
+        description: item.name || item.sku,
+        amount: Math.floor((item.final_price || item.price) * 100),
+        quantity: item.quantity
+      })
+    })
+
+    data.order = {
+      id: orderId,
+      amount: Math.floor(finalAmount * 100),
+      items: ipItems,
+      delivery_details: {
+        email: data.email,
+        name: data.customer.first_name + ' ' + data.customer.last_name,
+        phone_number: `${data.customer.phone_number}`,
+        address: {
+          line1: to.street + ', ' + String(to.number) || 's/n',
+          line2: to.complement || '',
+          city: to.city,
+          state: to.province || to.province_code,
+          zip: to.zip,
+          country: 'BR'
+        }
+      }
+    }
+
+    infiniteAxios
+      .then((axios) => {
+        // url: 'https://cloudwalk.github.io/infinitepay-docs/#autorizando-um-pagamento',
+        const headers = {
+          Accept: 'application/json'
+        }
+        if (isSandbox) {
+          headers.Env = 'mock'
+        }
+        return axios.post('/v2/transactions', data, { headers })
+      })
+      .then((response) => {
+        const { data } = response.data
+        const { attributes } = data
+        const intermediator = {
+          transaction_id: attributes.nsu,
+          payment_method: params.payment_method
+        }
+        if (attributes.authorization_id) {
+          console.log('Authorized transaction in InfinitePay')
+          intermediator.transaction_code = attributes.authorization_id
+          transaction.status = {
+            current: 'authorized',
```

Comment:
```suggestion
            current: 'paid',
```

Como não tem webhook, tem que entrar direto como pago...